samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -81,30 +92,41 @@ public IncomingMessageEnvelope(SystemStreamPartition systemStreamPartition, Stri
    * @param key A deserialized key received from the partition offset.
    * @param message A deserialized message received from the partition offset.
    * @param size size of the message and key in bytes.
-   * @param eventTime the timestamp (in epochMillis) of when this event happened
-   * @param arrivalTime the timestamp (in epochMillis) of when this event arrived to (i.e., was picked-up by) Samza
+   * @param eventTimeInstant the Instant when this event happened
+   * @param arrivalSystemNanoTime the System nanoTime when this event arrived to (i.e., was picked-up by) Samza
    */
   public IncomingMessageEnvelope(SystemStreamPartition systemStreamPartition, String offset,
-      Object key, Object message, int size, long eventTime, long arrivalTime) {
+      Object key, Object message, int size, Instant eventTimeInstant, long arrivalSystemNanoTime) {
     this(systemStreamPartition, offset, key, message, size);
-    this.eventTime = eventTime;
-    this.arrivalTime = arrivalTime;
+    this.eventTimeInstant = eventTimeInstant;
+    this.arrivalSystemNanoTime = arrivalSystemNanoTime;
+  }
+
+  /**
+   * Getter for eventTimeInstant
+   * @return Instant
+   */
+  public Instant getEventTimeInstant() {
```

Comment:
Do you need another getter for this?
Maybe you could recreate the `Instant` using `Instant.ofEpochMilli`, or mark `getEventTime` as deprecated to be replaced by `getEventTimeInstant`.

Reply:
marked getEventTime as Deprecated